controllers/imageupdateautomation_controller.go

```diff
@@ -229,7 +239,7 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		}
 	}
 
-	log.V(debug).Info("cloned git repository", "gitrepository", originName, "ref", ref, "working", tmp)
+	debuglog.Info("cloned git repository", "gitrepository", originName, "ref", ref, "working", tmp)
```

Comment:
Wouldn't it be better to log this before the attempt, so that it automatically adds context to any error that may happen during the attempt?

Reason I think this is a better approach, is because reconcilers generally do 3 things while performing operations for a resource:

1. Logging (to tell what is about to happen, or highlight important information during an action).
2. Events, to make certain things that have happened easily accessible to users and machines for observation.
3. Conditions, to advertise observed state. Most often recorded after a reconciler has run an action that changed state, or when it inspected it.

There is some overlap in 1 and 2, but what distinguishes it for me is that 2 in all scenarios I have seen thus far, it never happens before something has taken place, which means that it would be a better task for logging.

Reply:
Yes it would, and that is a well-reasoned insight :+1: